include/machine.h

```diff
@@ -10,6 +12,17 @@
 #include <machine/registerset.h>
 #include <hardware.h>
 
+#if defined(__CHERI_PURE_CAPABILITY__)
+#include <mode/cheri.h>
+/*  We trust the kernel to only use this powerful KernelVirtOffsetCap only
+ * when it needs to convert from physical to virtual addresses (and vice
+ * versa) and access them later on e.g., when iterating over page tables.
+ */
+#define KERNEL_OFFSET KernelVirtOffsetCap
+#else
+#define KERNEL_OFFSET PPTR_BASE_OFFSET
```

Comment:
This seems wrong. `KERNEL_OFFSET` is defined as a symbol in linker scripts:

https://github.com/seL4/seL4/blob/85aa104eb4d527bc9f0aac28f3e9c3a84eed1d7a/src/arch/arm/common_arm.lds#L15

Reply:
It was just a coincidence I went for clashing name (though has no effect on building or runtime). I'll change to another name